ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -152,234 +179,216 @@ public KsqlResource(
     this.ksqlEngine = ksqlEngine;
     this.replayableCommandQueue = replayableCommandQueue;
     this.distributedCommandResponseTimeout = distributedCommandResponseTimeout;
-    this.registerKsqlStatementTasks();
   }
 
   @POST
   public Response handleKsqlStatements(final KsqlRequest request) {
-    final List<PreparedStatement> parsedStatements;
-    final KsqlEntityList result = new KsqlEntityList();
-
     try {
-      parsedStatements = ksqlEngine.parseStatements(request.getKsql());
-      checkPersistentQueryCapacity(parsedStatements, request.getKsql());
-    } catch (final ParseFailedException e) {
-      return Errors.badStatement(e.getCause(), e.getSqlStatement(), result);
-    } catch (final KsqlException e) {
-      return Errors.badRequest(e);
-    }
+      final List<PreparedStatement<?>> statements = parseStatements(request.getKsql());
 
-    final Map<String, Object> streamsProperties = request.getStreamsProperties();
+      final Map<String, Object> propertyOverrides = request.getStreamsProperties();
 
-    for (final PreparedStatement parsedStatement : parsedStatements) {
-      final String statementText = parsedStatement.getStatementText();
-      final Statement statement = parsedStatement.getStatement();
-
-      try {
-        validateStatement(result, statementText, statement, streamsProperties);
-      } catch (final KsqlRestException e) {
-        throw e;
-      } catch (final KsqlException e) {
-        return Errors.badStatement(e, statementText, result);
-      } catch (final Exception e) {
-        return Errors.serverErrorForStatement(e, statementText, result);
-      }
+      validateStatements(statements, propertyOverrides);
```

Comment:
This is a significant change to the validation and error-handling logic. Previously, we'd validate each statement before executing it. If there were any errors with a given statement, we'd return an error with all the previous results in the body (in KsqlEntityList). This patch changes that quite a bit:
    1. We do the validation of all the statements together up front (which I think is fine, just including for completeness).
    2. If we fail when executing a statement, we drop all the previous results. Before, we'd include them in the error response.
    3. If we get a KsqlException, we don't return a statement error anymore. There is lots of validation that happens in the guts of the engine that raises KsqlException on bad statements (for example, using a non-key column in your join criteria). We should include the statement for those errors.

I think we should go back to validating and executing the statements one at a time (even if we still validate them all before executing them all), which lets us preserve the statement on KsqlException, and lets us build a partial response when a statement fails to execute but previous statements succeeded.

Reply:
Hey @Rohan,

1. Yep, switching to a more atomic style of validation / execution made more sense to me - but I think this should be done in a separate PR. So I'll revert it for now.

2. I'm not sure I follow. The code still collects a list of the previous results and include them in the error message when executing.  Do you mean we don't do this for validation? It wouldn't make sense to do it for validation, right?  Not that it matters for this PR as I'm going to revert.

3. What makes you say we don't return a statement error anymore? Do you mean 
`executeStatement` is not catching `KsqlException` and returning a `Errors.badStatement` style response? I'd of thought the validation would have resulted in the `KsqlException` / `Errors.badStatement`.  Though,  I guess there are edge cases where things can change between validation and execution that can result in issue.  I'll add that catch block in.  If this is not what you mean, then please elaborate.

Reply:
I've raised another issue to track the later work:  https://github.com/confluentinc/ksql/issues/2198

Reply:
FYI, on point 3, while I've reverted the change to pre-validate I've added back the code to include the partial results collected so far as part of validation failures. But I've only done this to the degree that the old code did, i.e. unknown, Queries and PrintTopic commands.

If I was expecting this to be a long standing change, then I'd wire it up more completely. However, I expect to be enhancing this to handle pre-validation very soon, so I think just maintaining the partial implementation that was there previously is enough.